src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java

```diff
@@ -98,14 +101,21 @@ public CaseDataValidator(final List<BaseTypeValidator> validators) {
                                                        final String fieldIdPrefix,
                                                        final BaseType fieldType) {
         return validators.stream()
-            .filter(validator -> validator.getType() == fieldType)
-            .findAny()
-            .map(baseTypeValidator -> baseTypeValidator
-                .validate(dataFieldId, dataValue, caseFieldDefinition)
-                .stream()
-                .map(result -> new ValidationResult(result.getErrorMessage(), fieldIdPrefix + result.getFieldId()))
-                .collect(Collectors.toList()))
-            .orElseThrow(() -> new RuntimeException("System error: No validator found for " + fieldType.getType()));
+            .filter(validator ->
+                validator.getType() == fieldType || isPredefinedTypeBaseValidator(validator, caseFieldDefinition.getFieldTypeDefinition().getId())
+            )
+            .map(baseTypeValidator -> baseTypeValidator.validate(dataFieldId, dataValue, caseFieldDefinition))
+            .flatMap(validationResults -> validationResults.stream()
+                .map(validationResult -> new ValidationResult(validationResult.getErrorMessage(), fieldIdPrefix + validationResult.getFieldId()))
+            ).collect(Collectors.toList());
+    }
```

Comment:
We can execute  all sorted validators and always take the first elements from the result, that can lead in more testing but it is generic enough